Add "guildwars2.com" to yellowlist

queicherius opened the pull request:
Privacybadger is blocking access to `api.guildwars2.com` which is the [official public API for the MMORPG Guild Wars 2](https://api.guildwars2.com/v2). It can only be used to read personal data off an account if the user previously generated and provided a valid API key (see "Authentication" section at the bottom), which makes this opt-in.

(Probably?) since all sites making use of that API are hosted on different domains, and most of the sites access the API from the client side, Privacybadger blocks the access to this API, regularly causing questions like [this](https://www.reddit.com/r/Guildwars2/comments/53zunu/gw2efficiency_account_values_statistics_update/d7xvuzk/?context=3) or [this](https://www.reddit.com/r/Guildwars2/comments/91j9lh/best_conversion_rates_for_zephyrite_supply_boxes/e300s4o/?context=2).

ghostwords commented:
I think you want to add the domain to the [yellowlist](https://github.com/EFForg/privacybadger/blob/ce4a8beedcb27079185d8f50cc6b54d3a2471d62/doc/fixing-broken-sites.md#external-domains-too-complex-to-surrogate-or-replace-with-placeholders), not the [multi-domain first-parties list](https://github.com/EFForg/privacybadger/blob/ce4a8beedcb27079185d8f50cc6b54d3a2471d62/doc/fixing-broken-sites.md#domains-that-are-part-of-the-site-but-dont-look-like-it).

ghostwords commented:
Could you provide the debugging information requested in https://github.com/EFForg/privacybadger/blob/1cac5ba752517802d0596645c464ffd55a60c9d0/.github/ISSUE_TEMPLATE/broken-site-report.md?

ghostwords commented:
We might want to add all of `guildwars2.com` ... See all these other subdomains below.

Error report counts by page domain and exact blocked "guildwars" subdomain:
```
+---------------------+------------------------+-------+
| fqdn                | blocked_fqdn           | count |
+---------------------+------------------------+-------+
| gw2efficiency.com   | api.guildwars2.com     |     8 |
| gw2efficiency.com   | render.guildwars2.com  |     6 |
| metabattle.com      | api.guildwars2.com     |     5 |
| help.guildwars2.com | guildwars2.zendesk.com |     4 |
| metabattle.com      | render.guildwars2.com  |     4 |
| qtfy.eu             | render.guildwars2.com  |     2 |
| deso.website        | api.guildwars2.com     |     1 |
| gw2crafts.net       | api.guildwars2.com     |     1 |
| gw2timer.com        | api.guildwars2.com     |     1 |
| qtfy.eu             | api.guildwars2.com     |     1 |
| snowcrows.com       | api.guildwars2.com     |     1 |
| gw2crafts.net       | render.guildwars2.com  |     1 |
| snowcrows.com       | render.guildwars2.com  |     1 |
| gw2timer.com        | tiles.guildwars2.com   |     1 |
| dps.report          | wiki.guildwars2.com    |     1 |
| dulfy.net           | wiki.guildwars2.com    |     1 |
| gw2efficiency.com   | wiki.guildwars2.com    |     1 |
| qtfy.eu             | wiki.guildwars2.com    |     1 |
| account.arena.net   | www.guildwars2.com     |     1 |
...
```

Error report counts by date and exact blocked "guildwars" subdomain:
```
+---------+------------------------+-------+
| ym      | blocked_fqdn           | count |
+---------+------------------------+-------+
| 2018-07 | api.guildwars2.com     |     3 |
| 2018-07 | render.guildwars2.com  |     3 |
| 2018-06 | api.guildwars2.com     |     1 |
| 2018-06 | render.guildwars2.com  |     1 |
| 2018-05 | api.guildwars2.com     |     2 |
| 2018-05 | render.guildwars2.com  |     2 |
| 2018-04 | api.guildwars2.com     |     1 |
| 2018-04 | tiles.guildwars2.com   |     1 |
| 2018-02 | guildwars2.zendesk.com |     1 |
| 2018-02 | render.guildwars2.com  |     1 |
| 2018-02 | wiki.guildwars2.com    |     1 |
...
```

bcyphers commented:
Possibly related to https://github.com/EFForg/privacybadger/issues/1545?

queicherius commented:
> I think you want to add the domain to the yellowlist, not the multi-domain first-parties list.

Ah, yes. I updated the PR to use the yellowlist instead.

> Could you provide the debugging information requested in  ...

Sadly not, I don't use PrivacyBadger, I just get reports of my page (gw2efficiency.com) not working fairly frequently, so I thought I'd fix the issue here. :)

> We might want to add all of guildwars2.com ... See all these other subdomains below.

Ah, yes. I added the TLD. Here is what the specific domains do, in case that's interesting:

- `api.guildwars2.com` is the URL of the public API endpoints
- `render.guildwars2.com` is for item icons and some other sprites linked in the API (essentially a CDN)
- `tiles.guildwars2.com` is for map tiles linked in the API (essentially a CDN)
- `wiki.guildwars2.com` is the official wiki (not really part of the API, but is hotlinked from some pages)
- `guildwars2.zendesk.com` and `www.guildwars2.com` have nothing to do with the API
